### PR TITLE
New searchbox, refactoring, fixes

### PR DESCRIPTION
**Pull request by Devas**
Isbns are calculated using title hashcode if they are missing
Added an evaluation of bookItem before adding it to booksToSend map;
Added tests for newly added method
Scrappers now return books even when they throw exception
Added exception handling to the App, added tests for new methods;
Added exception handling for google book converter;
Added localdatetime converter, the database now holds timestamp for entities
-REST Controller for searching
-Tests for REST controllers
-Searchbox
-Bugfix with sorting
-Bugfix with promo details display in a single book view

### CODE REVIEW

File: crawler/src/main/java/com/javaacademy/crawler/App.java [hunk 1]
```diff
@@ -49,7 +56,7 @@ void runApp(Set<Scrapper> scrappers) {
     private void logResults() {
         for (String storeName :
                 crawlersSendBooksStats.keySet()) {
-            statistics.info(() -> "Total books from " + storeName + ": " +crawlersSendBooksStats.get(storeName));
+            statistics.info(() -> "Total books from " + storeName + ": " + crawlersSendBooksStats.get(storeName));
```
Comment (patkkka) [inline crawler/src/main/java/com/javaacademy/crawler/App.java]: 👍

File: crawler/src/test/java/com/javaacademy/crawler/common/converters/GoogleBookConverterTest.java [hunk 2]
```diff
@@ -13,13 +12,33 @@
...
+        String title = "title";
+        long expectedIsbn = title.hashCode();
+        long isbn = googleBookConverter.generateIsbn(title);
+        assertEquals(isbn, expectedIsbn);
+    }
+
+    @Test
+    public void generateIsbnTestNegative() {
+        GoogleBookConverter googleBookConverter = new GoogleBookConverter();
+        String title = "Zarys ginekologii onkologicznej Tom 1-2";
```
Comment (piotr921) [inline crawler/src/test/java/com/javaacademy/crawler/common/converters/GoogleBookConverterTest.java]: 🎖

File: crawler/src/test/java/com/javaacademy/crawler/common/converters/GoogleBookConverterTest.java [hunk 3]
```diff
@@ -36,7 +55,7 @@ public void getIsbnTest() {
         Isbn isbn = mock(Isbn.class);
         isbnList.add(isbn);
         when(isbn.getIdentifier()).thenReturn("123123123123");
-        Long result = googleBookConverter.getIsbn(isbnList);
+        Long result = googleBookConverter.getIsbn(isbnList, "");
```
Comment (piotr921) [inline crawler/src/test/java/com/javaacademy/crawler/common/converters/GoogleBookConverterTest.java]: use String.EMPTY contant

File: rest-angular/src/app/main.component.html [hunk 4]
```diff
@@ -6,7 +28,7 @@
         <button class="col-md-2" (click)="sortByAuthor()">Author</button>
         <button class="col-md-2" (click)="sortByPrice()">Price</button>
         <button class="col-md-2" (click)="sortByPromo()">Promo</button>
-        <button class="col-sm-2" (click)="reset()">Reset</button>
+        <button class="col-sm-2" (click)="reset()">XD</button>
```
Comment (piotr921) [inline rest-angular/src/app/main.component.html]: 🎖

File: rest-angular/src/index.html [hunk 5]
```diff
@@ -3,15 +3,14 @@
 
 <head>
   <meta charset="utf-8">
-  <title>RestAngular</title>
+  <title>XD</title>
```
Comment (piotr921) [inline rest-angular/src/index.html]: ❌ D

File: crawler/src/test/java/com/javaacademy/crawler/common/converters/GoogleBookConverterTest.java (hunk 2 above)
Comment (patkkka) [inline crawler/src/test/java/com/javaacademy/crawler/common/converters/GoogleBookConverterTest.java]: 🥇

File: rest-angular/src/index.html (hunk 5 above)
Comment (patkkka) [inline rest-angular/src/index.html]: 🥇 🥇 🥇

**Pull request merged**